### PR TITLE
Update tutorials to use `create_agents` and `rng.integers`

### PR DESCRIPTION
**Pull request by DarshPareek**
### Summary
Updates the tutorials to align with Mesa 3.1's recommended practices for agent creation and random number generation.

### Motive
The changes make the tutorials consistent with Mesa 3.1 by:
- Using the `create_agents` class method instead of individual agent creation in loops
- Replacing `random.randrange` with `rng.integers` for position generation
- Improving performance by generating all agent positions at once instead of one at a time

### Implementation
Made consistent updates across `intro_tutorial.ipynb` and `MoneyModel.py`:

1. Agent Creation:
Old approach:
```python
for _ in range(self.num_agents):
    a = MoneyAgent(self)
```

New approach:
```python
agents = MoneyAgent.create_agents(model=self, n=n)
```

2. Position Generation:
Old approach:
```python
x = self.random.randrange(self.grid.width)
y = self.random.randrange(self.grid.height)
```

New approach:
```python
x = self.rng.integers(0, self.grid.width, size=(n,))
y = self.rng.integers(0, self.grid.height, size=(n,))
```

### Usage Examples
Example of the updated model initialization:
```python
class MoneyModel(mesa.Model):
    def __init__(self, n, width, height, seed=None):
        super().__init__(seed=seed)
        self.num_agents = n
        self.grid = mesa.space.MultiGrid(width, height, True)
        
        # Create agents all at once
        agents = MoneyAgent.create_agents(model=self, n=n)
        
        # Generate all positions at once
        x = self.rng.integers(0, self.grid.width, size=(n,))
        y = self.rng.integers(0, self.grid.height, size=(n,))
        
        # Place agents efficiently using zip
        for a, i, j in zip(agents, x, y):
            self.grid.place_agent(a, (i, j))
```

### CODE REVIEW

File: docs/tutorials/intro_tutorial.ipynb
Comment (EwoutH) [inline docs/tutorials/intro_tutorial.ipynb:1]: Let’s make they arguments explicit here:
```Python
create_agents(model=self, n=n)
```

**Review by EwoutH — COMMENTED** [review]
Thanks for the PR!

File: docs/tutorials/intro_tutorial.ipynb
Comment (quaquel) [reply docs/tutorials/intro_tutorial.ipynb:1]: I would also use self.rng to create the x and y coordinates

```python

x = self.rng.randint(0, self.grid.width, size=(n,))
y = self.rng.randint(0, self.grid.height, size=(n,))

```

**Comment by EwoutH** [comment]
I did assign @ashish-nagmoti to issue #2537 by the way. Next time please do check if an issue is assigned, and if it is, ask first.

@ashish-nagmoti would you like to review this PR instead?

File: docs/tutorials/intro_tutorial.ipynb
Comment (EwoutH) [reply docs/tutorials/intro_tutorial.ipynb:1]: Agreed, but other PR. Let’s keep them atomic, especially with new contributors.

**Comment by DarshPareek** [comment]
In the latest update I have changed the arguments to be more explicit in `create_agents` function and added 
```
x = self.rng.integers(0, self.grid.width, size=(n,)) 
y = self.rng.integers(0, self.grid.height, size=(n,))
```
to make positioning of agents better

File: docs/tutorials/intro_tutorial.ipynb
Comment (EwoutH) [reply docs/tutorials/intro_tutorial.ipynb:1]: @DarshPareek please don’t forget my original comment here.

File: docs/tutorials/MoneyModel.py [hunk 1]
```diff
@@ -62,12 +62,13 @@ def __init__(self, n=10, width=10, height=10, seed=None):
         self.grid = mesa.space.MultiGrid(width, height, True)
 
         # Create agents
-        for _ in range(self.num_agents):
-            a = MoneyAgent(self)
+        agents = MoneyAgent.create_agents(model=self,n=n)
```
Comment (EwoutH) [inline docs/tutorials/MoneyModel.py:65]: It’s best practice to leave a space after each argument when calling a method. Please update this everywhere.
```suggestion
        agents = MoneyAgent.create_agents(model=self, n=n)
```

Comment (DarshPareek) [reply docs/tutorials/MoneyModel.py]: I will format these according to black formatter

**Review by EwoutH — APPROVED** [review]
I’m good, but let’s see if Jan also is.

**Comment by quaquel** [comment]
I am fine with this, although I personally would prefer to move `self.grid.place_agent(a, (i, j))` into the `MoneyAgent`.

**Comment by EwoutH** [comment]
Personally I won't bother with updating the old spaces, we will replace them soon anyways.

**Comment by DarshPareek** [comment]
> Personally I won't bother with updating the old spaces, we will replace them soon anyways.
> 
> @DarshPareek Could you make your PR title and description a bit more clear/descriptive?

Thank you for the suggestion! I'll revise the PR title and description to better reflect the changes made. Please let me know if there’s anything specific you'd like included.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -32,3 +32,7 @@ repos:
         "--ignore-words",
         ".codespellignore",
       ]
+      exclude: >
```
Comment (EwoutH) [inline .pre-commit-config.yaml:35]: Sorry, I totally missed this in earlier reviews. Why are we excluding all notebooks from Codespell?

If there’s a certain word you would like to use, you can add it to [`.codespellignore`](https://github.com/projectmesa/mesa/blob/main/.codespellignore)

Comment (quaquel) [reply .pre-commit-config.yaml]: Could you please remove all changes to .codespellignore?

File: .codespellignore [hunk 3]
```diff
@@ -8,3 +8,10 @@ fpr
 falsy
 assertIn
 nD
+te
+Fo
+fo
+COO
+Onl
+Te
+ND
```
Comment (EwoutH) [inline .codespellignore:17]: Sorry, this is a bit nitpicky, but could you explain why all these “words” need to be added?

Comment (DarshPareek) [reply .codespellignore:17]: Okay so the codespell is flagging these words as errors they are present in the notebook, i think they are the part of image data that is used. 
Screenshot 2024-12-14 at 19-29-46
here "COO" is considered as a word so it it flagged by codespell and thus the branch fails the pre-commit test.
If there is another workaround this I will gladly update the branch

File: .pre-commit-config.yaml (hunk 2 above)
Comment (DarshPareek) [reply .pre-commit-config.yaml]: If i don't add words to .codespellignore the pre-commit check fails on the branch, If I remove them is there any workaround the check?

File: .codespellignore (hunk 3 above)
Comment (EwoutH) [reply .codespellignore:17]: Okay, that clears up a lot.

I would propose just cleaning the cell outputs.

**Comment by DarshPareek** [comment]
I have cleared the outputs and removed the words from .codespellignore, let me know if any further changes are needed.

File: .pre-commit-config.yaml (hunk 2 above)
Comment (EwoutH) [reply .pre-commit-config.yaml:35]: This still needs to be cleaned up

Comment (DarshPareek) [reply .pre-commit-config.yaml:35]: I have cleaned it up.

**Comment by EwoutH** [comment]
Cleanup the final things, merging.

Thanks @DarshPareek!

File: .pre-commit-config.yaml (hunk 2 above)
Comment (EwoutH) [reply .pre-commit-config.yaml]: Final things done in f4b7eb6.

**Review by EwoutH — APPROVED** [review]

**Comment by DarshPareek** [comment]
> Cleanup the final things, merging.
> 
> Thanks @DarshPareek!

Thanks! I would love to contribute in future.

**Pull request merged**